lib/tty/pager.rb

```diff
@@ -73,11 +86,51 @@ def enabled?
     # @return [TTY::Pager]
     #
     # @api public
-    def page(text, &callback)
-      pager.page(text, &callback)
+    def page(text)
+      pager.page(text)
+      self
+    end
+
+    # Write the given text to the available pager
+    #
+    # @param [Array<String>] *args
+    #   strings to send to the pager
+    #
+    # @raise [PagerClosed]
+    #   if the write failed due to a closed pager tool
+    #
+    # @return [TTY::Pager]
+    #
+    # @api public
+    def write(*args)
+      pager.write(*args)
+      self
+    end
+    alias_method :<<, :write
+
+    # Write a newline-ended line to the available pager
+    #
+    # @param [String] text
+    #   the text to send to the pager
+    #
+    # @return [TTY::Pager]
+    #
+    # @api public
+    def puts(text)
+      pager.puts(text)
       self
     end
 
+    # Close the pager tool, wait for it to finish
+    #
+    # @return [Boolean]
+    #   whether the pager exited successfully
+    #
+    # @api public
+    def close
+      pager.close
+    end
```

Comment:
How about we use `forwardable` module to delegate these?

Reply:
This is doable for `close`, although I do get this warning in tests: 

``` ruby
warning: TTY::Pager#close at /home/andrew/projects/tty-pager/lib/tty/pager.rb:132 forwarding to private method RSpec::Mocks::Double#close
```

The reason this happens is because the `pager` is a protected attribute, I think. Though replacing the `spy(:system_pager)` with a `spy(:system_pager, close: true)` no longer shows the error, so I'm not sure what the deal is.

For the other methods like `puts` and `write`, it's a bit trickier, because their interface is supposed to return `self`, but if we delegate, the `self` is going to be the inner pager, not the container. Which seems subtly counter-intuitive. It shouldn't be a practical problem, but it just means that `pager << "foo" << "bar"` changes its type halfway through, which is kind of weird.

To be honest, the relationship between `TTY::Pager` and the other ones is a bit weird. It's both their parent class, and also a container for pager instances. I'd consider separating their responsibilities -- have the `NullPager` be a parent, or some form of an abstract pager that has all the methods implemented but raises errors to ensure all the children implement it. Have the `TTY::Pager` class have `.page` with a block and `.get_pager(**options)` or something, and nothing else. But that's something for later.

Reply:
Ahh I see - never mind. Sorry for the trouble. Let's forget about my suggestion and revert back to what it was. Introducing a module for only one method doesn't seem to be worth it. 

Agreed, an interface/abstract class for all the three pager types would be good. This is what happens when you bolt-on new pagers over time and don't stop to refactor. It all started with a system pager only and then grew branches. Definitely would be interested in making this cleaner and potentially more extensible for when people want to implement their own paging logic or extend existing. However, this kind of change would need to wait for the next release. I'd rather have many releases than one big bang release. This will allow for feedback from potential users.